app/server/tests/test_custom_auth.py

```diff
@@ -0,0 +1,16 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, override_settings
+
+from .. import custom_auth
+
+@override_settings(HEADER_AUTH_PROXY_GROUPS='Admin')
+@override_settings(HTTP_X_AUTH_PROXY_USER='UserA')
```

Comment:
Is this required? If I read correctly, we're setting `HTTP_X_AUTH_PROXY_USER` to `UserA` which means that the client should authenticate via:

```sh
curl -H 'UserA: some-user-name' http://...
```

That seems incorrect to me.